examples/light-switch-app/nrfconnect/main/AppTask.cpp

```diff
@@ -0,0 +1,601 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "AppTask.h"
+#include "LEDWidget.h"
+#include "LightSwitch.h"
+#include "ThreadUtil.h"
+
+#include <platform/CHIPDeviceLayer.h>
+
+#include <lib/support/CHIPMem.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/SafeInt.h>
+
+#include <app/server/OnboardingCodesUtil.h>
+#include <app/server/Server.h>
+
+#include <credentials/DeviceAttestationCredsProvider.h>
+#include <credentials/examples/DeviceAttestationCredsExample.h>
+
+#ifdef CONFIG_CHIP_OTA_REQUESTOR
+#include <app/clusters/ota-requestor/BDXDownloader.h>
+#include <app/clusters/ota-requestor/DefaultOTARequestorStorage.h>
+#include <app/clusters/ota-requestor/GenericOTARequestorDriver.h>
+#include <app/clusters/ota-requestor/OTARequestor.h>
+#include <platform/nrfconnect/OTAImageProcessorImpl.h>
+#endif
+
+#include <dk_buttons_and_leds.h>
+#include <logging/log.h>
+#include <system/SystemError.h>
+#include <zephyr.h>
+
+using namespace ::chip;
+using namespace ::chip::app;
+using namespace ::chip::Credentials;
+using namespace ::chip::DeviceLayer;
+
+LOG_MODULE_DECLARE(app);
+namespace {
+constexpr EndpointId kLightSwitchEndpointId           = 1;
+static constexpr uint32_t kFactoryResetTriggerTimeout = 3000;
+static constexpr uint32_t kFactoryResetCancelWindow   = 3000;
+static constexpr uint32_t kDimmerTriggeredTimeout     = 500;
+static constexpr uint32_t kDimmerInterval             = 300;
+static constexpr size_t kAppEventQueueSize            = 10;
+
+K_MSGQ_DEFINE(sAppEventQueue, sizeof(AppEvent), kAppEventQueueSize, alignof(AppEvent));
+
+LightSwitch sLightSwitch;
+
+LEDWidget sStatusLED;
+LEDWidget sDiscoveryLED;
+LEDWidget sBleLED;
+LEDWidget sUnsedLED;
+
+static bool sIsThreadProvisioned    = false;
+static bool sIsThreadEnabled        = false;
+static bool sIsThreadBLEAdvertising = false;
+static bool sIsSMPAdvertising       = false;
+static bool sHaveBLEConnections     = false;
+static bool sIsDiscoveryEnabled     = false;
+static bool sWasDimmerTriggered     = false;
+
+k_timer sFunctionTimer;
+k_timer sDimmerPressKeyTimer;
+k_timer sDimmerTimer;
+
+#ifdef CONFIG_CHIP_OTA_REQUESTOR
+DefaultOTARequestorStorage sRequestorStorage;
+GenericOTARequestorDriver sOTARequestorDriver;
+OTAImageProcessorImpl sOTAImageProcessor;
+chip::BDXDownloader sBDXDownloader;
+chip::OTARequestor sOTARequestor;
+#endif
+
+} /* namespace */
+
+AppTask AppTask::sAppTask;
+
+CHIP_ERROR AppTask::Init()
+{
+    /* Initialize CHIP */
```

Comment:
It is a minor comment, but in general we use in the upstream cpp // comments style and in the NCS C /* */ style.